x/rewards/keeper/distribution.go

```diff
@@ -202,25 +205,16 @@ func (k Keeper) distributeBlockRewards(ctx sdk.Context, blockDistrState *blockRe
 			Add(contractDistrState.InflationaryRewards).
 			Add(contractDistrState.FeeRewards...)
 
-		if err := k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ContractRewardCollector, rewardsAddr, rewards); err != nil {
-			panic(fmt.Errorf("sending rewards (%s) to the rewards address (%s) for the contract (%s): %w", rewards, rewardsAddr, contractDistrState.ContractAddress, err))
-		}
-
-		// Emit distribution event
-		types.EmitContractRewardDistributionEvent(
-			ctx,
-			contractDistrState.ContractAddress,
-			rewardsAddr,
-			rewards,
-		)
+		// Create a new record
+		rewardsRecordState.CreateRewardsRecord(rewardsAddr, rewards, calculationHeight, calculationTime)
 	}
 }
 
 // cleanupTracking prunes all tracking data for the given block height for x/tracking and x/rewards modules.
 func (k Keeper) cleanupTracking(ctx sdk.Context, height int64) {
 	// We can prune the previous block ({height}), but that makes tracking CLI queries useless as there won't be any data.
 	// Pruning history block also makes e2e tests possible.
-	// TODO: this should be replaced with a param
+	// TODO: this should be replaced with a param?
 	heightToPrune := height - 10
```

Comment:
If I understood correctly, gastracking records are kept for 10 blocks?

Reply:
Yes, exactly. I thought it might be useful for someone to query gas and rewards tracking info for their contract. So we keep them for 10 blocks.